### DISCUSSION TITLE
Install in non-docker environment

### DISCUSSION THREAD

**Discussion by spusuf** [music-assistant/support#1391] [Q&A]
Is there a way to install the server in a non-dokcer environment? Docker has much higher resource requirements and idle pwoer consumption as compared to something like a BSD jail. Hence I am running HomeAssistant Core (without any issues) in a jail. Is there a way to install all the neccessary components manually? Or is it possible to use a python venv like home assistant.

**Comment by marcelveldt** [comment]
BSD is not a supported OS for Home Assistant, nor Music Assistant. In theory it will maybe work but you will get in trouble at some point and then you are on your own with support. The added overhead of docker is really low and not worth the trouble imo.

And yes, you can install in a venv just like Home Assistant and make sure you have ffmpeg installed:

pip install music-assistant[server]
and then simply run 'mass' to run the server

But again, doing so is on your own risk. I think you should just use the docker images which provide the optimized environment to run Music Assistant.

**Comment by spusuf** [comment]
Docker runs at 30% CPU utilisation at idle, where my current server with double the services (including the ones in docker) uses 5%. 

Everything installs without any errors, but when I run 
`python mass.py`
Python runs it without errors, then nothing happens, no error in console, no logs, no web interface. Is there any way to see logs and diagnose what's happening?

**Reply by marcelveldt** [reply]
I really don't believe that, sorry. Docker's overhead may be a couple of mb here and there and maybe 0.1% of cpu but nothing more. You have some serious other issues on your platform then. If it's a pi, why not load it with Home Assistant S ad call it a day.

Are you running on BSD ? if so, I think I have told you before that that is a NOT supported platform for both HA and MA.
If I were you, I'd first switch to a more common OS. If you dont wat the run the Home Assistant appliance (Home Assistant OS), you may want to look at debian with docker. This is a dead end and is going to leave you with no support and a lot of time wasted. Just my 2 cents

**Reply by spusuf** [reply]
Yes that's the overhead. It's not a pi, it's a dedicated server.

Yes I'm on BSD, like I said Home Assistant Core is running fine in a venv and it appears so is music assistant but I can't tell because there aren't any logs.

I'm not asking you to support this config, I'm asking where the logs are so I can do it myself. I don't want your OS recommendations. I'm not running a debian install just because one service isn't working. But since you seem so cynical I've got everything else working including home assistant, NGINX, uptime kuma, Plex, jellyfin,  Zoneminder, Photoprism, n8n, obsidian Livesync, and 2 Minecraft servers in mineos. The idle power consumption is 6w from the wall. The same hardware with only home assistant and NGINX would never go below 10% idle usage, averaging 35%, and drawing 25w from the wall. Again, all I'm asking for the location of the logs so I can do it myself.

**Reply by marcelveldt** [reply]
logs are printed to stdout and stored in in the datadir you have started the server with
